lib/schnorr.py

```diff
@@ -139,7 +139,7 @@ def sign(privkey, message_hash):
     if _secp256k1_schnorr_sign:
         sig = create_string_buffer(64)
         res = _secp256k1_schnorr_sign(
-            secp256k1.secp256k1.ctx, sig, message_hash, privkey, None, None
```

Comment:
Is the default argument of `b''` equivalent to the previous `None` here?

Reply:
Seems so from looking at libsec256k1 but I will retest to make sure the values don't change.

Reply:
I'm just worried about unintended side effects.  I believe `None` ends up in secp as a C `NULL` where as `b''` ends up in C as .. a string of size 0 e.g. `""`.

Why change the default behavior of this function?  Can't you have the `ndata` argument default to `None`?  And then for paycode it will work as you like and for secp it will work as you like.

Reply:
No. This is not equivalent.  I just verified the C side.  It's as I suspected. `None` here ends up as -> `NULL` in C.  Whether or not that pointer is `NULL` or `""` on the C side completely changes the behavior of the nonce function.

See this bit of code which is where the behavior is affected: https://github.com/Bitcoin-ABC/secp256k1/blob/69ccf0d2f758b1a4df9cde8222459381e08fc1dc/src/secp256k1.c#L330

That `ndata` ends up exactly as the `data` pointer in the C.  If the `data` pointer is not NULL, then **unconditionally** 32 bytes of data are appended (but that pointer points to 0 bytes!).

You actually should be enforcing that `len(ndata)` == 32, or that `ndata` is None..  all over this codebase as well.  This will lead to crashes.